TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/integ_test/integ_test/test_cases/ma_test_base.py

```diff
@@ -0,0 +1,108 @@
+        self.allow_source_target_combinations = allow_source_target_combinations or []
+        self.run_isolated = run_isolated
+        self.short_description = short_description
+        self.console_link_env = console_link_env
+        if ((not console_link_env.source_cluster or not console_link_env.target_cluster) or
+                (not console_link_env.source_cluster.version or not console_link_env.target_cluster.version)):
+            raise RuntimeError("Both a source cluster and target cluster must be defined for the console library and "
+                               "include the version field")
+        self.source_cluster = console_link_env.source_cluster
+        self.target_cluster = console_link_env.target_cluster
+        self.source_version = ClusterVersion(version_str=self.source_cluster.version)
+        self.target_version = ClusterVersion(version_str=self.target_cluster.version)
+
+        supported_combo = False
+        for (allowed_source, allowed_target) in allow_source_target_combinations:
+            if (is_incoming_version_supported(allowed_source, self.source_version) and
+                    is_incoming_version_supported(allowed_target, self.target_version)):
+                supported_combo = True
+                break
+        if not supported_combo:
+            raise ClusterVersionCombinationUnsupported(self.source_version, self.target_version)
+
+        self.source_operations = get_operations_library_by_version(self.source_version)
+        self.target_operations = get_operations_library_by_version(self.target_version)
+        self.console_config_path = console_config_path
+        self.unique_id = unique_id
+        self.snapshot: Snapshot = console_link_env.snapshot
+        self.backfill: Backfill = console_link_env.backfill
+        self.metadata: Metadata = console_link_env.metadata
+        self.replayer: Replayer = console_link_env.replay
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}(source={self.source_version},target={self.target_version})>"
+
+    def perform_initial_operations(self):
+        pass
+
+    def perform_snapshot_create(self):
+        snapshot_result: CommandResult = self.snapshot.create(wait=True)
+        assert snapshot_result.success
+
+    def perform_operations_after_snapshot(self):
+        pass
+
+    def perform_metadata_migration(self):
+        metadata_result: CommandResult = self.metadata.migrate()
+        assert metadata_result.success
+
+    def perform_operations_after_metadata_migration(self):
+        pass
+
+    def start_backfill_migration(self):
+        backfill_start_result: CommandResult = self.backfill.start()
+        assert backfill_start_result.success
+        # small enough to allow containers to be reused, big enough to test scaling out
+        backfill_scale_result: CommandResult = self.backfill.scale(units=2)
+        assert backfill_scale_result.success
+
+    def perform_operations_during_backfill_migration(self):
```

Comment:
What does it mean to run 'during'?  Will you run continuously?  I suspect that this is really just guaranteed to run before the next phase is run and after the previous one has finished - which isn't quire the same as during since a backfill might not have started backfilling or it could have been done.